test/playlist.test.js

```diff
@@ -385,11 +385,92 @@ function(assert) {
 
   assert.equal(seekable.start(0), 0, 'starts at the earliest available segment');
   assert.equal(seekable.end(0),
-              9 - (2 + 2 + 1),
-              'allows seeking no further than three segments from the end');
+              9 - (2 + 2 + 1 + 2),
+              'allows seeking no further than three times target duration from the end');
   assert.equal(playlistEnd, 9, 'playlist end at the last segment');
 });
 
+QUnit.test('safeLiveIndex returns the correct media index for the safe live point',
+function(assert) {
```

Comment:
It might be wise to break this up into 3 tests; 

`safe live point for standard durations`, `correct media index for variable durations`, `no safe live point`.  

This will make it easier to debug (it's very clear which test is failing), and I believe that the following assertions  won't get hit if one assertion fails.  As an example, if `standard durations` fails, but `variable segments` passes that is useful debug information (or if both fail).  I think that gets lost with this though.

Reply:
Previous assertions failing won't prevent the following from being run, tested your example and variable segments still gets hit and passes. I can still break them up if you think its a good idea, but all the assertions in this test are testing the same function, just different input edge cases, so to me it makes sense to keep all the `safeLiveIndex` assertions within the same test

Reply:
I think keeping the tests kind of small/specific will make maintenance longer term a bit simpler.  Particularly if another case gets tacked on here.